src/pages/index.js

```diff
@@ -43,83 +116,134 @@ profileEditButton.addEventListener("click", () => {
+//   return cardElement.getNewCard();
+// }
+
+ 
+function getCard(items) {
+  const cardElement = new Card(
+    items,
+    "#card-template",
+    handleImagePreview,
+    handleCardDeleteSubmit,
+    handleLikeClick
+  );
 
-const addCardFormValidator = new FormValidator(addCardEditForm, options);
+  return cardElement.getNewCard();
+}
 
-const addCardModal = new PopupWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit
-);
 
-/* ----------------------- */
-/*     Image Preview      */
-/* ----------------------*/
+function handleLikeClick(items) {
+  //toggle the like status
+  api.cardLikeStatus(items._id,items._isLiked).then(() => {
+    items.updateLikes(items._isLiked);
+  }).catch((err) => {
+    console.error("Failed to update card likes status", err);
+    items.updateLikes(items._isLiked); //revert the like status if failed to update api
+  });
+}
 
-const previewImageModal = new PopupWithImage({
-  modalSelector: "#preview-image-modal",
-});
-previewImageModal.setEventListeners();
+// function handleLikeClick(items) {
+//   const isLiked = !items.isLiked;
+//   if (isLiked === true) {
+//     api.cardLikeStatus(items._id, isLiked).then(() => {
+//       items.updateLikes(true);
+//     }).catch((err) => {
+//       console.error(res);
+//     });
+//   } else {
+//     api.cardUnlikeStatus(items._id).then(() => {
+//       items.updateLikes(false); 
+//   }).catch((err) => {
+//     console.error(err);
+//   });
+//   console.log(items);
+//   }
+// }
+
+function handleAvatarSubmit({ avatarUrl }){
+  console.log(avatarUrl);
+  editAvatarPopup.setLoading(true, "Saving...");
+  api.patchProfileAvatar(avatarUrl)
+  .then((data) => {
+    profileUserInfo.setUserAvatar(data.avatar);
+    editAvatarPopup.close();
+  })
+  .finally(()=>{
+    editAvatarPopup.setLoading(false);
+  }).catch((err)=> {
+    console.error(err);
+  })
+}
 
-function handleImagePreview(cardData) {
-  previewImageModal.open(cardData.name, cardData.link);
+function handleImagePreview(card) {
+  previewImageModal.open(card.name, card.link);
 } // it needs name and link to display image on preview
 
-/* ----------------------- */
-/*     Form Validation     */
-/* ----------------------- */
-// const formValidators = {};
-
-// const enableValidation = (options) => {
-//   const formList = Array.from(document.querySelectorAll(options.formSelector));
-//   formList.forEach((formElement) => {
-//     const validator = new FormValidator(formElement, options);
-//     const formName = formElement.getAttribute("name");
-//     formValidators[formName] = validator;
-//     validator.enableValidation();
-//   });
-// };
-// enableValidation(options);
+//create a function that handle delete button modal
+function handleCardDeleteSubmit(items){
+  deleteSubmitConfirmModal.open();
+  deleteSubmitConfirmModal.setSubmit(()=> {
+    deleteSubmitConfirmModal.setLoading(true, "Deleting...");
+    this._cardElement.dataset.id = this._id;
```

Comment:
`this` should **not** be used **outside** classes in functions in `index.js` . This **context** is easy to lose. You need to find other ways of passing data.